frontend/src/search/hooks/useSearch.ts

```diff
@@ -0,0 +1,107 @@
+import { useState, useEffect } from 'react';
+import { useSearchFoldersQuery, useSearchJournalsQuery } from '@/graphql/generated';
+
+const useSearch = () => {
+  const [searchQuery, setSearchQuery] = useState('');
+  const [isSearching, setIsSearching] = useState(false);
+  const [inputValue, setInputValue] = useState('');
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const [activeFilter, setActiveFilter] = useState<string | null>(null);
+
+  const {
+    data: folderData,
+    loading: folderLoading,
+    error: folderError,
+  } = useSearchFoldersQuery({
+    variables: { searchQuery },
+    skip: !searchQuery,
+  });
+
+  const {
+    data: journalData,
+    loading: journalLoading,
+    error: journalError,
+  } = useSearchJournalsQuery({
+    variables: { searchQuery },
+    skip: !searchQuery,
+  });
+
+  const isLoading = folderLoading || journalLoading;
+  const error = folderError || journalError;
+
+  useEffect(() => {
+    if (inputValue) {
+      const debounceSearch = setTimeout(() => {
+        setSearchQuery(inputValue);
+        setIsSearching(true);
+      }, 3000);
```

Comment:
I feel 3 seconds is a bit too long for this. 
Maybe something like 500ms or 1s would be a more responsive experiecne?

Reply:
Yeah, Im thingking about that too Haha

Reply:
I changed it to 500ms

Reply:
Done!